pypicloud/cache/base.py

```diff
@@ -140,10 +139,13 @@ def upload(
         if old_pkg is not None and not self.allow_overwrite:
             raise ValueError("Package '%s' already exists!" % filename)
         if self.calculate_hashes:
-            file_data = data.read()
```

Comment:
I couldn't find where this function is used, but I refactored it to re-use `stream_file` (and avoid whole file in memory)